### ISSUE TITLE
Fixes to prevent hanging on zombie children have made test for children to all be dead fail intermittently

### ISSUE THREAD

**Issue by adamnovak** [DataBiosphere/toil#4024]
As @mr-c noted, we have a flaky test here now:

https://ucsc-ci.com/databiosphere/toil/-/jobs/8639#L5368

```
________________ SingleMachineBatchSystemTest.testProcessEscape ________________
Traceback (most recent call last):
  File "/builds/databiosphere/toil/src/toil/test/batchSystems/batchSystemTest.py", line 665, in testProcessEscape
    fcntl.lockf(lockfile, fcntl.LOCK_EX | fcntl.LOCK_NB)
BlockingIOError: [Errno 11] Resource temporarily unavailable
```

_Originally posted by @mr-c in https://github.com/DataBiosphere/toil/issues/4022#issuecomment-1024890016_

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1137)
┆friendlyId: TOIL-1137

**Comment by adamnovak** [comment]
@w-gao THis is almost certainly related to #3974 and the fix for #3965. We now don't hang around until these child processes are dead enough to unlock their files before returning from the batch system shutdown, so the files appear locked still.

Unless we start looping through process group members somehow and making sure they are all either gone or waiting to be reaped, I'm not sure how we fix this 100%. Maybe we want to bring back that kill loop but with a timeout this time? Or maybe we do really want to enumerate processes in the group?

**Comment by w-gao** [comment]
Ah, [testProcessEscape()](https://github.com/DataBiosphere/toil/blob/master/src/toil/test/batchSystems/batchSystemTest.py#L570) expects children to all be dead after batch system shutdown, but #3974 removes that guarantee.

I wonder if we can just add a slight delay after we send each SIGKILL [here](https://github.com/DataBiosphere/toil/blob/master/src/toil/batchSystems/singleMachine.py#L324), since normal processes should exit fairly quickly with SIGKILL? To be safe, we can also bring back the kill loop but with a timeout.

> Or maybe we do really want to enumerate processes in the group?

Would a sigkill signal to the process group reach all child processes? If not, we could enumerate the processes in the group and send a kill signal individually, but I think we still need to wait a while for the signal to propagate.
